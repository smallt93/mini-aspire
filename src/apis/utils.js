import isObject from 'lodash/isObject';
import isArray from 'lodash/isArray';

export const handleError = (error) => {
  const { response, request, message } = error;
  if (response) {
    const { data, status } = response;
    if (isObject(data)) {
      const errorList = Object.keys(data).map(key => isArray(data[key]) ? data[key][0] : data[key]);
      return { error: errorList, errorStatus: status };
    }

    if (isArray(data)) return { error: data, errorStatus: status };

    // Request made and server responded
    if (data.includes('<!DOCTYPE html>')) {
      return {
        error: `[${status}] Unexpected error`,
        errorStatus: status,
      };
    }
    return { error: `[${status}] ${data}`, errorStatus: status };
  }
  if (request) {
    // The request was made but no response was received
    return { error: 'Unable to send your request' };
  }
  // Something happened in setting up the request that triggered an Error
  return { error: message };
};
