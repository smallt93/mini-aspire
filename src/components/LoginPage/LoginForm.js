import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Formik, Form } from 'formik';
import * as Yup from 'yup';
import FormInput from '../FormInput/FormInput';
import { LoginButton } from './LoginPage.style';

const validationSchema = Yup.object().shape({
  email: Yup.string().trim().required('Invalid email'),
  password: Yup.string().trim()
    .required('Password is required'),
});

const initialValues = {
  email: '',
  password: '',
};

export default class LoginForm extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
  }

  static propTypes = {
    login: PropTypes.func,
    isLoading: PropTypes.bool,
    errorMessage: PropTypes.any,
  }

  loginSubmit = (loginValues) => {
    const { login } = this.props;
    login(loginValues);
  }

  renderLoginBtn = () => {
    const { isLoading } = this.props;
    return (
      isLoading
        ? <LoginButton primary>Loading ...</LoginButton>
        : <LoginButton type="submit">Login</LoginButton>
    );
  }

  render() {
    return (
      <Formik
        initialValues={initialValues}
        validationSchema={validationSchema}
        onSubmit={this.loginSubmit}
      >
        <Form noValidate>
          <FormInput
            type="email"
            name="email"
            label="Email"
          />
          <FormInput
            type='password'
            name="password"
            label="Password"
          />
          {this.renderLoginBtn()}
          
          <Link to="/register">
            <LoginButton register>Register</LoginButton>
          </Link>
        </Form>
      </Formik>
    );
  }
}
