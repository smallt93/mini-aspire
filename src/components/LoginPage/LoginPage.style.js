import styled, { css, keyframes } from 'styled-components';

const BounceInput = keyframes`
 0% {
    transform: translateX(0px);
    timing-function: ease-in;
  }
  37% {
    transform: translateX(14px);
    timing-function: ease-out;
  }
  55% {
    transform: translateX(-14px);
    timing-function: ease-in;
  }
  73% {
    transform: translateX(14px);
    timing-function: ease-out;
  }
  82% {
    transform: translateX(-14px);
    timing-function: ease-in;
  }
  91% {
    transform: translateX(14px);
    timing-function: ease-out;
  }
  96% {
    transform: translateX(-14px);
    timing-function: ease-in;
  }
  100% {
    transform: translateX(0px);
    timing-function: ease-in;
  }
`;
export const LoginPageWrap = styled.div``;

export const LoginMessage = styled.div``;

export const LoginItemWrap = styled.div`
  width: 100%;
  height: 100vh;
  max-width: 480px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  ${({ noFlex }) => noFlex && css`
    display: block;
    height: auto;
  `}

  .react-tabs__tab-list {
    display: flex;
    margin-bottom: 1em;
    cursor: pointer;
    .react-tabs__tab {
      flex: 1;
      text-align: center;
      padding: .75em 2em;
      border: 1px solid #ccc;
      &:first-child {
        border-right: none;
        border-radius: 5px 0px 0px 5px;
      }
      &:last-child {
        border-radius: 0px 5px 5px 0px;
      }
      &:hover {
        color: #fff;
        background-color: #02b358;
        border-color: #02b358;
      }
    }
    .react-tabs__tab--selected {
      color: #fff;
      background-color: #00d366;
      border-color:#00d366;
    }
  }
`;

export const LoginItem = styled.div`
  padding: 2em;
  border: 1px solid #ccc;
  border-radius: 5px;
`;

export const LoginTitle = styled.div`
  margin: 10px 0px 40px;
  text-align: center;
  h1 {
    font-size: 28px;
    font-family: 'MontserratBold';
    line-height: 35px;
    color: ${props => props.theme.colorStyled.ColorPrimary};
    margin: 20px 0px 5px;
  }
  p {
    font-size: 16px;
    line-height: normal;
    font-family: 'MontserratLight';
    color: #7f7f7f;
  }
  @media (max-width: 480px) {
    text-align: center;
  }
`;

export const LoginButton = styled.button`
  width: 100%;
  min-height: 35px;
  padding: 10px 130px;
  border-radius: 8.5px;
  color: ${props => props.theme.colorStyled.ColorWhite};
  border: 1px solid ${props => props.theme.colorStyled.ColorBgDefault};
  background-color: ${props => props.theme.colorStyled.ColorBgDefault};
  box-shadow: 0px 4px 6.5px 0 #ff9e1630;
  font-size: 15px;
  margin-bottom: 15px;
  outline: none;
  cursor: pointer;
  &:hover {
    color: ${props => props.theme.colorStyled.ColorWhite};
    border-color: #a3a0a0;
    background-color: #a3a0a0;
    box-shadow: 0 0 10.5px 0 rgba(0, 0, 0, 0.06);
  }
  ${({ primary }) => primary && css`
    color: ${props => props.theme.colorStyled.ColorPrimary};
    border-color: ${props => props.theme.colorStyled.ColorWhite};
    background-color: ${props => props.theme.colorStyled.ColorWhite};
    box-shadow: 0 0 10.5px 0 rgba(0, 0, 0, 0.06);
  `}
  ${({ register }) => register && css`
    color: ${props => props.theme.colorStyled.ColorWhite};
    border-color: ${props => props.theme.colorStyled.ColorMing};
    background-color: ${props => props.theme.colorStyled.ColorMing};
    box-shadow: 0 0 10.5px 0 rgba(0, 0, 0, 0.06);
  `}
  @media (max-width: 1440px) {
    padding: 10px 60px;
  }
  @media (max-width: 480px) {
    padding: 10px 20px;
  }
`;

export const LoginAction = styled.div``;
