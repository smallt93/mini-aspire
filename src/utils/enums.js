export const ROLE_TYPE = {
  USER: 'user',
  ADMIN: 'admin',
};
