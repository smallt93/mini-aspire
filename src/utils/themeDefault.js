export const defaultThemes = {
  fontSize: {
    XSmallMobileFontSize: '0.570em',
    SmallMobileFontSize: '0.645em',
    DefaultFonts: '0.715em',
    SmallerFontSize: '0.785em',
    SmallFontSize: '0.8125em',
    MediumFontSize: '0.8570em',
    PrimaryFontSize: '0.9285em',
    DefaultFontSize: '1em',
    LargeFontSize: '1.070em',
    LargerFontSize: '1.145em',
    HeadingH5FontSize: '1.215em',
    HeadingH4FontSize: '1.285em',
    HeadingH3FontSize: '1.360em',
    HeadingH2FontSize: '1.430em',
    HeadingH1FontSize: '1.570em',
    XLargeFontSize: '1.715em',
  },
  borderRadius: {
    borderbtn: '0.325em',
    borderBlock: '0.6em',
  },
  colorStyled: {
    ColorWhite: '#ffffff',
    ColorBlack: '#000000',
    ColorBgDefault: '#00d366',
    ColorBgDefaultHover: '#02b358',
    ColorBgSilver: '#f4f5f7',
    ColorBgHover: '#ffe2ba',
    ColorBgContent: '#f6f7fb',
    ColorPrimary: '#2b1a08',
    ColorSecond: '#a7a7a7',
    ColorDark: '#1b1b1b',
    ColorLighterGrey: '#999999',
    ColorDarkerGrey: '#666666',
    ColorGrey: '#6e7079',
    ColorYellowCheckbox: '#ffc700',
    ColorRedDel: '#d72111',
    ColorLightGrey: '#bdb8b6',
    ColorBtn: '#f5f5f5',
    ColorBtnHover: '#eaeaea',
    ColorBtnDark: '#e2e2e2',
    ColorBorder: '#e7e7e7',
    ColorBorderInput: '#091e4214',
    ColorHeaderTable: '#585754',
    ColorBgLeft: '#f4f5f7',

    ColorPrimaryGreen: '#1bd18a',
    ColorPrimaryBlue: '#4886ff',
    ColorPrimaryBlueHover: '#3778f7',
    ColorBgMyCourses: '#fbfbfb',
    ColorBgSideBarDefault: '#f9f9f9',
    ColorDefault: '#6f6f6f',
    ColorSuccess: '#1bd18a',
    ColorSuccessN: '#37921e',
    ColorDarkGrey: '#020b0d',
    ColorMidGrey: '#e8e8e8',
    ColorTime: '#aeaeae',

    ColorXBorder: '#e0dedb',
    ColorPoint: '#a5a5a5',
    ColorBorderClassNav: '#edeff1',

    ColorXLightGrey: '#1e2328',
    ColorXXLightGrey: '#2a2f34',
    ColorXXXLightGrey: '#0c0d12',
    ColorLink: '#433aef',
    ColorActiveCheckbox: '#18c798',
    ColorLevelKill: '#8c9093',
    ColorLevelKillActive: '#1787e2',
    ColorBgClassNav: '#f7f9fa',
    ColorBoxShadow: '#dbdbdb',
    ColorSilver: '#a4a4a4',
    ColorWarming: '#5160ef',
    ColorPurple: '#6669C9',
    ColorBgForm: '#ececec',
    ColorSkipped: '#bdbdbd',
    ColorMing: '#385e7e',
    ColorPeach: '#fee2b4',
    ColorBgAvartar: '#d2d2d2',
    ColorUser: '#1a1a1a',
    ColorWarning: '#ed1c24',

    ColorBorderSilver: '#cbcbcb',
    ColorBorderPractice: '#dedede',
    ColorBorderHampton: '#e3d0ad',
    ColorSilverChalice: '#9e9e9e',
    ColorBgSeashell: '#f7f7f7',
    ColorAlto: '#d6d6d6',
    ColorHint: '#d9d9d9',
    ColorAlizarinCrimson: '#D92222',
    ColorFunGreen: '#01a41c',
    ColorDoveGray: '#6e6e6e',

    ColorGreenPoint: '#34DF9D',
    ColorRedPoint: '#ba0c28',
    ColorSelectedScoreSeperate: 'rgba(163,173,186,0.2)',
    ColorSectionDashboard: '#767C84',
    ColorTopicUpIcon: '#415063',
    ColorTableHeader: '#0c0d12',
    ColorScoreStep: '#0f1013',
    ColorLightCoral: '#ff7f7f',
    ColorLightOrange: '#fff2e3',
    ColorOrange: '#ff9e16',
    ColorLightRedGray: '#f2eceb',
    ColorRedGray: '#453635',
    ColorLightYellow: '#fff9e3',
    ColorYellow: '#ffce2a',
    ColorMidBlue: '#528ffd',

    ColorSkilLevelOne: '#d11010',
    ColorSkilLevelTwo: '#69cc27',
    ColorQuiteBlue: '#4A86E8',
  },
};
