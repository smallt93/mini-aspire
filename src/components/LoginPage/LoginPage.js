import React, { Component } from 'react';
import PropTypes from 'prop-types';

import LoginForm from '../../containers/LoginPage/LoginForm';
import {
  LoginPageWrap,
  LoginMessage,
  LoginItemWrap,
  LoginItem,
  LoginTitle,
} from './LoginPage.style';

class LoginPage extends Component {
  static propTypes = {
    statusMessage: PropTypes.string,
  }

  renderStatusMessage = () => {
    const { statusMessage } = this.props;
    return (
      <LoginMessage>{statusMessage}</LoginMessage>
    );
  }

  render() {
    return (
      <LoginPageWrap>
        {this.renderStatusMessage()}
        <LoginItemWrap>
          <LoginItem>
            <LoginTitle>
              <h1>Welcome to Aspire</h1>
            </LoginTitle>
           
            <LoginForm />
          </LoginItem>
        </LoginItemWrap>
      </LoginPageWrap>
    );
  }
}

export default LoginPage;