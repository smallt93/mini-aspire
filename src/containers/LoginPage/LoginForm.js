import { connect } from 'react-redux';
import { withRouter} from 'react-router-dom';
import LoginForm from '../../components/LoginPage/LoginForm';

const mapStateToProps = () => ({});

const mapDispatchToProps = {};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(LoginForm));