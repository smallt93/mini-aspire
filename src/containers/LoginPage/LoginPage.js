import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import LoginPage from '../../components/LoginPage';

const mapStateToProps = () => ({});

const mapDispatchToProps = {};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(LoginPage));