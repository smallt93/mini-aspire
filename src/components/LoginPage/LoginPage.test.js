import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter as Router } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import LoginForm from './LoginForm';
import { LoginButton } from './LoginPage.style';
import { defaultThemes } from '../../utils/themeDefault';

describe('<LoginForm />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const loginMock = jest.fn();

  const Component = mount(
    <Router>
      <ThemeProvider theme={defaultThemes}>
        <LoginForm
          login={loginMock}
          isLoading={false}
          errorMsg={null}
        />
      </ThemeProvider>
    </Router>
    ,
  );

  it('should render', () => {
    expect(Component.length).toBe(1);
  });
  it('match snapshot', () => {
    expect(Component).toMatchSnapshot();
  });

  describe('<form />', () => {
    it('form element should have a onSubmit attribute', async () => {
      await (() => {
        expect(Component.props().onSubmit).toBeDefined();
      })
    });

    it('onSubmit attribute should be of type `function`', async () => {
      await (() => {
        expect(typeof Component.props().onSubmit === 'function').toBe(true);
      })
    });

    it('form element should have an `<input />` element', async () => {
      await (() => {
        expect(Component.find('form').childAt(0).type()).toBe('input');
      })
    });

    describe('<input />', () => {
      it('input element should be of type `text`', async () => {
        await (() => {
          expect(Component.find('form').childAt(0).props().type).toBe('text');
        })
      });

      it('input element value should be empty', async () => {
        await (() => {
          expect(Component.find('form').childAt(0).props().value).toBe('');
        })
      });

      it('input element should have an onChange attribute', async () => {
        await(() => {
          expect(Component.find('form').childAt(0).props().onChange).toBeDefined();
        })
      });

      it('onChange attribute should be of type `function`', async () => {
        await(() => {
          expect(typeof Component.find('form').childAt(0).props().onChange === 'function').toBe(true);
        })
      });

      it('should update the state when a value is email input', async () => {
        await(() => {
          const name = 'nmtri931@gmail.com';
          const input = Component.find('form').childAt(0);
          input.simulate('change', {
            target: {
              name: 'email',
              value: name,
            }
          });
          expect(Component.state().fields.name).toBe(name);
        })
      });

      it('should update the state when a value is password input', async () => {
        await(() => {
          const pw = '12345678';
          const input = Component.find('form').childAt(0);
          input.simulate('change', {
            target: {
              name: 'password',
              value: pw,
            }
          });
          expect(Component.state().fields.name).toBe(pw);
        })
      });
    });
  });

  it('should call loginMock once when Login button is clicked', async () => {
    await (() => {
      // trigger useEffect
      act(() => Component );
  
      const LoginBtnComp = Component.find(LoginButton).at(0);
      LoginBtnComp.simulate('click');
      expect(loginMock).toBeCalledTimes(1);
    })
  });
});
