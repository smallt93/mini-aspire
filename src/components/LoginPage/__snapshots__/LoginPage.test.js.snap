// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> match snapshot 1`] = `ReactWrapper {}`;
